### PR TITLE
Z-Wave compliance changes to generic handlers. NOV-570

### PR DESCRIPTION
**Pull request by mckeed**
Improve Z-Wave spec conformance for generic catch-all handlers.

### CODE REVIEW

File: devicetypes/smartthings/zwave-controller.src/zwave-controller.groovy [hunk 1]
```diff
@@ -43,34 +43,59 @@ metadata {
...
+def zwaveEvent(physicalgraph.zwave.commands.crc16encapv1.Crc16Encap cmd) {
+	createEvent(isStateChange: true, descriptionText: "$device.displayName: ${cmd.encapsulatedCommand()}")
 }
 
 def zwaveEvent(physicalgraph.zwave.Command cmd) {
-	def event = [isStateChange: true]
-	event.linkText = device.label ?: device.name
-	event.descriptionText = "$event.linkText: $cmd"
-	event
+	createEvent(isStateChange: true, "$device.displayName: $cmd"
```
Comment (bflorian) [inline devicetypes/smartthings/zwave-controller.src/zwave-controller.groovy]: Looks like something's missing here?

**Review by bflorian — CHANGES_REQUESTED** [review]

File: devicetypes/smartthings/zwave-controller.src/zwave-controller.groovy [hunk 2]
```diff
@@ -40,37 +40,68 @@ metadata {
...
+
+private command(physicalgraph.zwave.Command cmd) {
+	if (deviceIsSecure) {
+		zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+	} else {
+		cmd.format()
+	}
+}
+
+private deviceIsSecure() {
```
Comment (dkirker) [inline devicetypes/smartthings/zwave-controller.src/zwave-controller.groovy]: This will need to be getDeviceIsSecure for line 94 to work.

Comment (dkirker) [inline devicetypes/smartthings/zwave-controller.src/zwave-controller.groovy]: This will need to be getDeviceIsSecure for line 94 to work.

**Review by dkirker — CHANGES_REQUESTED** [review]

File: devicetypes/smartthings/zwave-basic-smoke-alarm.src/zwave-basic-smoke-alarm.groovy [hunk 3]
```diff
@@ -173,6 +177,18 @@ def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd, results
 	results << createEvent(map)
 }
 
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd, results) {
+	def encapsulatedCommand = cmd.encapsulatedCommand([ 0x80: 1, 0x84: 1, 0x71: 2, 0x72: 1 ])
+	state.sec = 1
```
Comment (dkirker) [inline devicetypes/smartthings/zwave-basic-smoke-alarm.src/zwave-basic-smoke-alarm.groovy:182]: I've noticed a use of state.sec as 1 and as true. Would setting to a boolean make more sense for consistency?

**Review by dkirker — CHANGES_REQUESTED** [review]

File: devicetypes/smartthings/zwave-basic-smoke-alarm.src/zwave-basic-smoke-alarm.groovy (hunk 3 above)
Comment (mckeed) [reply devicetypes/smartthings/zwave-basic-smoke-alarm.src/zwave-basic-smoke-alarm.groovy:182]: It looks like it's using 1 and 0 pretty much everywhere. I fixed one place in this PR that was using true.

File: devicetypes/smartthings/zwave-controller.src/zwave-controller.groovy (hunk 2 above)
Comment (mckeed) [reply devicetypes/smartthings/zwave-controller.src/zwave-controller.groovy]: fixed

File: devicetypes/smartthings/zwave-basic-smoke-alarm.src/zwave-basic-smoke-alarm.groovy (hunk 3 above)
Comment (dkirker) [reply devicetypes/smartthings/zwave-basic-smoke-alarm.src/zwave-basic-smoke-alarm.groovy:182]: Perfect!! Works for me. :)

**Review by dkirker — APPROVED** [review]

**Comment by workingmonk** [comment]
@mckeed can you add a ticket number to the PR title?

**Pull request merged**